HP-161: 관리자 페이지 - 구독 상품, 회원 관리, 상품 관리 데이터 패칭

jieun419 opened the pull request:
## PR 타입

- [x] 기능 추가
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 문서 수정
- [x] 기타

## 반영 브랜치

<!-- 예: feature/login → develop -->

`HP-161` → `dev`

## 작업 사항

- 관리자 페이지 
   - 구독 상품 관리 GET 요청, pagination/select 기능 추가
   - 회원 관리 GET 요청, pagination/select 기능 추가
   - 상품 관리  GET 요청, pagination 기능 추가

**[기타]**
- format 유틸 함수 추가
   - 날짜 데이터 -> yyyy/mm/dd 형식으로 반환
   - 가격 데이터 -> 1000 -> 1,000으로 반환
-  admin/management/member  -> admin/management/user 로 path 변경

**[공유 사항]**
- 자주 변경되지 않고, 재 호출을 방지 하기 위해 카테고리 데이터에 캐싱 옵션을 추가 했습니다. (1시간 정도)
- 검색 기능은 api 작어 중인 것 같아 완료 되는대로 추가하겠습니다.
- 이미지 데이터가 엑박 떠서 이미지 데이터 정상화 후 table ui 다시 확인해야하 것 같습니다.
- header ui 수정 등은 따로 티켓 만들어서 수정하겠습니다.
- 구독상품관리 데이터의 경우 목업 데이터 요청 했으나.. 대기하는데 시간이 걸릴 것 같아 우선 push 합니다.

## 체크리스트

- [x] 빌드에 성공했나요?
- [x] 코드 컨벤션을 잘 지켰나요? (예: `cmd + opt + L`)

## 테스트 결과

|구독상품관리|회원관리|상품관리|
|-|-|-|
|<img width="1415" height="814" alt="스크린샷 2025-07-28 오후 6 07 06" src="https://github.com/user-attachments/assets/c072766d-920f-4d61-ba83-8e1891f9ff5c" />|<img width="1416" height="804" alt="스크린샷 2025-07-28 오후 6 03 05" src="https://github.com/user-attachments/assets/3f7a477f-aca9-49f8-929e-d6738af0e175" />|<img width="1416" height="805" alt="스크린샷 2025-07-28 오후 6 04 40" src="https://github.com/user-attachments/assets/6275dc73-5f1d-4756-86d1-6a26647f916a" />|

jieun419 committed:
Feat: adminManagement 관련 api, query, path, type 설정

jieun419 committed:
Refactor/Design: Table 관련 props 및 스타일 수정

jieun419 committed:
Feat: router 설정

jieun419 committed:
Feat: 구독 상품 관리 데이터 패칭 및 관련 유틸 함수 추가
- format 함수 추가
- ADMIN_SELECT_ITEMS 데이터 추가
- 구독 상품 관리 데이터 패칭

jieun419 committed:
Feat: 회원 관리 페이지 패칭

jieun419 committed:
Feat: 상품 관리 데이터 패칭

jieun419 committed:
Refactor/Chore: 기타 코드 수정
- AdminContainer : 스타일 수정
- LayoutContainer : className 옵셔널 타입으로 수정
- AdminNavigation : path 및 변수 이름 수정

jieun419 committed:
Chore: 불필요한 코드 삭제

jieun419 committed:
Merge branch 'dev' into HP-161/je

jieun419 committed:
Chore: 코드 충돌 수정 하면서 import 순서 변경

minseung-gang commented on src/components/table/ui/TableCell.tsx:
```diff
@@ -28,16 +26,52 @@ const TableCell = ({ className, attribute = 'td', type = 'text', children }: Tab
 
-  const tableCellClass = `${className} ${getAttributeClass()} ${getTypeClass()}`;
+  const getTextAlignClass = () => {
+    if (type !== 'text') return;
```
안녕하세요 지은님!
혹시 그냥 return문을 사용하는 경우 className에서 undefined를 return에서 dom에서 undefined 문자열이 적히지는 않을까요? 혹시 우려되어 말씀드렸습니다.
혹시나 제가 잘못알고 있는 거라면 잘못 말씀드려 죄송합니다!

jieun419 replied:
좋은 지적입니다!
전혀 죄송하실 필요 없다 생각합니다. 이런 피드백 언제나 환영입니다! 😊

type이 text가 아닌 경우 빈 문자열을 반환하도록 수정했습니다!

`if (type !== 'text') return '';` 


[Fix/Refactor: TableCell 컴포넌트 오타 및 예외처리 강화](https://github.com/happy-pill/happyPill_Front/pull/14/commits/1f40b61417ef776df79d5d275e326fd0502581c0)

minseung-gang commented on src/components/table/ui/TableCell.tsx:
```diff
@@ -28,16 +26,52 @@ const TableCell = ({ className, attribute = 'td', type = 'text', children }: Tab
+
+    switch (textAlign) {
+      case 'center':
+        return 'item-center text-center';
```
items-center를 item-center로 잘못 적으신것 같습니다!

jieun419 replied:
다음엔 오타 검수 확장자 사용해 보겠습니다ㅠㅠ
오타 확인 감사합니다!


[Fix/Refactor: TableCell 컴포넌트 오타 및 예외처리 강화](https://github.com/happy-pill/happyPill_Front/pull/14/commits/1f40b61417ef776df79d5d275e326fd0502581c0)

minseung-gang commented on src/components/table/ui/TableCell.tsx:
```diff
@@ -28,16 +26,52 @@ const TableCell = ({ className, attribute = 'td', type = 'text', children }: Tab
+    }
+  };
+
+  const getFlexTypeClass = () => {
```
혹시 switch문에 default case가 없으면 undefined가 반환되는 거 아닐까요?
그럼 위에 제가 말씀드린 것과 같이 undefined가 리턴 될 것 같습니다!

jieun419 replied:
수정했습니다! 감사합니다!

[Fix/Refactor: TableCell 컴포넌트 오타 및 예외처리 강화](https://github.com/happy-pill/happyPill_Front/pull/14/commits/1f40b61417ef776df79d5d275e326fd0502581c0)

minseung-gang commented on src/pages/adminManagementSubscribe/components/molecule/SelectBox.tsx:
```diff
@@ -0,0 +1,30 @@
+import { ADMIN_SELECT_ITEMS } from '@/constants/common';
+
+interface SelectBoxProps {
+  crrentSize: number;
```
currentSize를 잘못 적으신 것 같습니다!

jieun419 replied:
수정 했습니다! 감사합니다!

[Fix/Refactor: 구독 상품,회원,상품 관리 오타 및 상수 데이터 컨벤션에 맞게 수정](https://github.com/happy-pill/happyPill_Front/pull/14/commits/f9c1772623addec13b4d0804d7af079afd9c0078)

minseung-gang commented on src/pages/adminManagementSubscribe/components/atom/atom/Title.tsx:
Title 컴포넌트의 폴더 위치가 조금 이상한 것 같아요!
components/atoms/Title.tsx가 맞는 것 같습니다!

jieun419 replied:
헛 페이지 하나 다 만들고 하나씩 폴더를 복사 붙여 놨는데 그 때 문제가 생겼나 봅니다ㅠ
확인 감사합니다!

[Fix: Title 폴더 구조 수정](https://github.com/happy-pill/happyPill_Front/pull/14/commits/ec4a2265eb7afceffd040a8581444f09997bf019)

minseung-gang commented on src/pages/adminManagementSubscribe/components/molecule/SubscribeTable.tsx:
```diff
@@ -0,0 +1,36 @@
+      </Table.Header>
+      <Table.Body>
+        {subscribeList.contents.map((item) => (
+          <Table.Row key={item.notifyEmail}>
```
key를 email로 적어주셨는데,
혹시 저희가 사용되는 알림받을 이메일이 중복되는 경우가 있지는 않을까요?
그럼 key가 유일키가 아닌 게 될 것 같아서 한번 말씀드려봤습니다!

jieun419 replied:
subscriptionId 필드가 존재하는데
상품 관련 id라 착각한 것 같습니다ㅠ 확인 감사합니다!

[Fix/Refactor: 구독 상품,회원,상품 관리 오타 및 상수 데이터 컨벤션에 맞게 수정](https://github.com/happy-pill/happyPill_Front/pull/14/commits/f9c1772623addec13b4d0804d7af079afd9c0078)

minseung-gang commented on src/pages/adminManagementSubscribe/index.tsx:
```diff
@@ -0,0 +1,45 @@
+
+      <TopControlsSection crrentSize={size} onChange={onChangeSize} />
+
+      {subscribeList && subscribeList?.contents.length > 0 ? (
```
&& 연산자로 subscriptList를 체크하고 있는데, subscibeList?.content.length라고 적어주셔서 중복되는 것 같아요!

jieun419 replied:
adminList 관련 데이터는 page,totalPage 등의 데이터도 같이 반환합니다!

- contents데이터가 빈 배열이여도 page,totalPage 데이터가 항상 존재해 데이터가 있다 판단하게 됩니다.
- subscribeList?.contents.length > 0라는 조건문만 추가할 경우
`'subscribeList.contents.length'은(는) 'undefined'일 수 있습니다.` 라는 메세지를 표시해서

따라서  `subscribeList && subscribeList?.contents.length > 0` 형식으로 더블 체크했습니다!

다만 subscribeList 있는지 이미 체크하고 있어서 `?` 옵셔널 연산자를 삭제 했습니다!

[Fix/Refactor: 구독 상품,회원,상품 관리 오타 및 상수 데이터 컨벤션에 맞게 수정](https://github.com/happy-pill/happyPill_Front/pull/14/commits/f9c1772623addec13b4d0804d7af079afd9c0078)

minseung-gang replied:
아! 네 지은님 말씀이 맞습니다! 
삼항연산자를 통해서 로직을 실행시키려면 undefined나 null은 적절하지 않죠! 
제가 깊게 생각하지 못했던 것 같습니다!
타입안정성을 위해 &&은 함께 사용하는게 맞네요!

minseung-gang commented on src/pages/adminManagementSubscribe/components/molecule/SubscribeTable.tsx:
```diff
@@ -0,0 +1,36 @@
+import Table from '@/components/table/ui/Table';
+import { formatDateToFullDateSlide } from '@/utils/format';
+
+const tableHeader = ['상품이름', '알림받을 이메일', '주문번호', '배송일'];
```
상수로 유지되는 값을 담은 변수 네이밍은 대문자로 적는게 맞지 않을까 생각되어 말씀드려봅니다!

jieun419 replied:
수정했습니다! 감사합니다!

[Fix/Refactor: 구독 상품,회원,상품 관리 오타 및 상수 데이터 컨벤션에 맞게 수정](https://github.com/happy-pill/happyPill_Front/pull/14/commits/f9c1772623addec13b4d0804d7af079afd9c0078)

minseung-gang commented on src/pages/adminManagementUser/components/molecule/SelectBox.tsx:
```diff
@@ -0,0 +1,30 @@
+import { ADMIN_SELECT_ITEMS } from '@/constants/common';
+
+interface SelectBoxProps {
+  crrentSize: number;
```
currentSize를 적으시려고 하셨던 걸까요?? 오타인 것 같습니다

minseung-gang commented on src/pages/adminManagementUser/index.tsx:
```diff
@@ -0,0 +1,42 @@
+
+      <TopControlsSection crrentSize={size} onChange={onChangeSize} />
+
+      {userList && userList?.contents.length > 0 ? (
```
이 부분도 전에 말씀드린 것과 같이 옵셔널 체이닝을 사용해주셨는데 &&와 중복이 되는 것 같습니다.

jieun419 replied:
subscribeList 관련 답변과 동일합니다!

minseung-gang commented on src/pages/adminManagementUser/components/organism/UserListSection.tsx:
```diff
@@ -0,0 +1,27 @@
+
+import Pagination from '@/components/pagination/Pagination';
+
+interface userListSectionProps {
```
네이밍 일관성을 위해서는 첫글자가 대문자로 쓰여야되는데 소문자로 잘못 들어가 있습니다!

minseung-gang commented on src/pages/adminManagementUser/components/molecule/UserTable.tsx:
```diff
@@ -0,0 +1,57 @@
+              />
+            </Table.Cell>
+            <Table.Cell>{formatDateToFullDateSlide(item.createdAt)}</Table.Cell>
+            <Table.Cell>{formatDateToFullDateSlide(item.deletedAt)}</Table.Cell>
```
deletedAt 부분에 null을 체크하는 부분이 필요할 것 같습니다!
AdminUser 타입에서 deletedAt은 string타입으로 지정되어 있지만
삭제되지 않은 사용자는 deletedAt이 없을 것 같은데
그럼 혹시 null이나 undefined로 오는게 아닌 걸까요..?
이 부분에 대해서 한번 더 체크가 필요할 것 같습니다!
만약에 null이 들어오는 경우에 컴포넌트에서 체크를 하거나 함수 내부에서 체크하는 부분이 필요할 것 같아요!

jieun419 replied:
수정했습니다! 감사합니다!

[Refactor: format 유틸 함수 반환 값 예외 처리](https://github.com/happy-pill/happyPill_Front/pull/14/commits/6791567fd51d712d26e260053de70e067d155bae)

minseung-gang commented on src/pages/adminManagementUser/components/molecule/UserTable.tsx:
```diff
@@ -0,0 +1,57 @@
+            <Table.Cell>{formatDateToFullDateSlide(item.deletedAt)}</Table.Cell>
+            <Table.Cell type='button'>
+              {item.isDeleted ? (
+                <Button variant='orange' size='S'>
```
여기 부분에 Button 컴포넌트는 onClick핸들러는 나중에 api가 구현이되면 추가해주시는 부분인걸까요?

jieun419 replied:
넵 맞습니다!
해당 브랜치는 단순 GET api 관련된 기능만 적용습니다.

추후 각 페이지 POST api 관련 작업 시 적용 예정입니다!

minseung-gang commented on src/pages/adminManagementProduct/components/molecule/ProductTable.tsx:
```diff
@@ -0,0 +1,58 @@
+import { useGetCategoryDetail } from '@/hooks/api/member/category';
+import { formatValueToComma } from '@/utils/format';
+
+const tableHeader = ['썸네일', '이름', '제조사', '간단설명', '카테고리', '가격', '재고', ''];
```
이 부분도 상수값들을 저장하는 변수의 네이밍은 대문자로 작성해주시면 좋을 것 같습니다!

minseung-gang commented on src/pages/adminManagementProduct/index.tsx:
```diff
@@ -0,0 +1,40 @@
+
+      <TopControlsSection />
+
+      {productList && productList?.contents.length > 0 ? (
```
이 부분도 전과 같이 옵셔널체이닝과 &&이 중복되어 반복되는 패턴인것 같습니다.

minseung-gang commented on src/pages/adminManagementProduct/index.tsx:
```diff
@@ -0,0 +1,40 @@
+
+const Index = () => {
+  const [page, setPage] = useState(1);
+  const { data: productList } = useGetProductList(page, Math.min(...ADMIN_SELECT_ITEMS));
```
혹시 adminManagementProduct 페이지에서는 다른 페이지들과 달리 size를 state로 관리하지않고 바로 상수로 넣어주신 이유가 있으실까요??

jieun419 replied:
adminManagementProduct 페이지는 size 기능이 없고 "상품 등록" 버튼만 존재합니다.
다른 페이지와 일관성을 유지하기 위해 `ADMIN_SELECT_ITEMS`의 최소 값으로 지정했습니다! (최소 리스트 10개로)

추후 초기 리스트 개수 변경이 필요할 경우
ADMIN_SELECT_ITEMS 데이터만 수정하면 admin 관련 모든 리스트 초기 `size`가 동일하게 됩니다!

+) 어쩌면 피그마 UI 수정이 안된 걸 수도 있겠네요... 이부분은 피그마로 구조 고민이 필요할 수도 있겠습니다ㅠ

<img width="880" height="185" alt="스크린샷 2025-07-29 오후 3 11 52" src="https://github.com/user-attachments/assets/19a7555a-369c-4744-bf89-dc62c8841727" />

minseung-gang commented on src/types/common.ts:
```diff
@@ -1,3 +1,5 @@
   locale: 'ko' | 'en';
 }
+
+export type Provider = 'GOOGLE' | 'KAKAO';
```
저번에 지은님께서 LocaleType타입이 type이 아니라 interface로 일관성을 유지해야한다고 말씀주셨던 것 같습니다 ㅠㅠ
Provider도 type에서 일관성을 위해 interface로 수정해주시는게 맞을 것 같습니다!

jieun419 replied:
민승님께 오해의 소지를 드린 것 같아 죄송합니다ㅠ

제가 말씀드렸던 `interface로 일관성 유지`는
props 타입, api 데이터 반환 타입 등, 객체 형식의 경우 interface를 사용하자는 거였습니다!

현재 처럼 LocaleType, Provider 유형은 type을 사용해도 좋다 마인드입니다!

우선 Provider 타입을 하나로 합쳤습니다!

[Refactor: Provider 타입 삭제 및 AdminUser 타입에 바로 적용](https://github.com/happy-pill/happyPill_Front/pull/14/commits/3073d8d56050b52e1c52e93d3c3adeebc64be331)

jieun419 committed:
Fix/Refactor: TableCell 컴포넌트 오타 및 예외처리 강화

jieun419 committed:
Fix: Title 폴더 구조 수정

jieun419 committed:
Fix/Refactor: 구독 상품,회원,상품 관리 오타 및 상수 데이터 컨벤션에 맞게 수정
- tableHeader -> TABLE_HEADER
- crrentSize -> currentSize
- **List?.contents 옵셔널 제거
- 구독 상품 관리) key 값 notifyEmail -> subscriptionId로 변경

jieun419 committed:
Refactor: format 유틸 함수 반환 값 예외 처리

jieun419 committed:
Refactor: Provider 타입 삭제 및 AdminUser 타입에 바로 적용

jieun419 committed:
Merge branch 'dev' of https://github.com/happy-pill/happyPill_Front into HP-161/je
